### ISSUE TITLE
Feature request: MIDI note based triggers

### ISSUE THREAD

**Issue by chanson6680** [smeighan/xLights#3228]
**Is your feature request related to a problem? Please describe.**
There are not many options for MIDI note based triggers. The only way I have found to do this is using the piano function. however Piano is extremely limited in what you can do with it.

**Describe the solution you'd like:**
I would like the ability to trigger timing events specified by note, EG similar to "timing track fade" or "Timing tack toggle" but be able to specify a MIDI note to trigger. Note: for this to work well you would probably also have to specify trigger on note on or trigger on note off.

**Describe alternatives you've considered**
I have used the piano function and chroma keyed it to make it light up certain lights for certain events, however it will only turn the light on for the moment the MIDI not is "ON" there are no options for fade out etc.

**Additional context**
Basically I just want more note based MIDI controls :) It would really make programming so much nicer when midi tracks are available.  Here is what I have so far: notes roughly trigger vertical strips(not entirely accurately) and I am unable to fade off lights. https://drive.google.com/file/d/1YAIPNbP31do1anudi-TSUrlOoOj3UuGv/view?usp=sharing

**Comment by keithsw1111** [comment]
Have you tried using a piano timing track along with the state effect?

**Comment by chanson6680** [comment]
Yes, However this still does not give me the customization tools I would like. The ability to do midi based note triggers in the "value curve" pop up dialogs would be ideal and give the most tools for customization. Here is what I am Envisioning(see pics) I photoshopped this together
![Midi Original](https://user-images.githubusercontent.com/77990478/168483565-9e8ed307-515a-472f-b38e-99001d86704e.png)
![Midi Modified](https://user-images.githubusercontent.com/77990478/168483568-a23f841a-e78d-495b-b644-4a4166281de2.png)

**Comment by derwin12** [comment]
Dont see this being implemented... just a bit too niche.
